### ISSUE TITLE
Shrink developer and data engineering policies

### ISSUE THREAD

**Issue by ewastempel** [ministryofjustice/modernisation-platform#6232]
### User Story

As an MP engineer
I need to be able to swiftly respond to user requests when it comes to adding permissions to developer and data engineering policy
So that I can unblock users in a swift manner.

### Value / Purpose

This is to support MP scalability.

### Additional Information

In short, data engineering policy is made of developer policy and additional permissions and recently we were hitting policy size quota limits. This ticket is to shrink the size of both developer and data engineering policies in order to be able to cope with the quota limits.
See this [thread](https://mojdt.slack.com/archives/C013RM6MFFW/p1708010445158539) for more info.

### Proposal / Unknowns

See where you can use `*` to reduce a list of permissions without jeopardising the permissiveness of the policy.

### Definition of Done

- [ ] Developer and data engineering policies are updated
- [ ] Changes are deployed to sprinkler and tested
- [ ] Changes are deployed to the rest of member accounts
- [ ] Pipeline runs successfully when changes are merged

**Comment by julialawrence** [comment]
I've been eyeing up the DE policy for a while as a candidate for review so I'll raise a corresponding ticket in our backlog.

**Comment by ep-93** [comment]
https://github.com/ministryofjustice/modernisation-platform/pull/6256/files

This would complete the ticket, but there has been a change of plan


```perhaps we could remove the dependency on the developer role? The developer role itself isn’t hitting the size limit quota yet

**Comment by ep-93** [comment]
Julia had a great idea in this thread too. https://moj.enterprise.slack.com/archives/C013RM6MFFW/p1708354195372729

Great ideas, but dont solve the immediate problem. My PR was approved, this ticket is solved, will add a link to a new ticket to follow Julias ideas.
